inc/class-wpseo-custom-fields.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO
+ */
+
+/**
+ * WPSEO_Custom_Fields
+ */
+class WPSEO_Custom_Fields {
+	/**
+	 * @var array Cache the custom fields.
+	 */
+	protected static $custom_fields = null;
+
+	/**
+	 * Retrieves the custom field names as an array.
+	 *
+	 * @see WordPress core: wp-admin/includes/template.php. Reused query from it.
+	 *
+	 * @return array The custom fields.
+	 */
+	public static function get_custom_fields() {
+		global $wpdb;
+
+		// Use cached value if available.
+		if ( ! is_null( self::$custom_fields ) ) {
+			return self::$custom_fields;
+		}
+
+		/**
+		 * Filters the number of custom fields to retrieve for the drop-down
+		 * in the Custom Fields meta box.
+		 *
+		 * @since 2.1.0
```

Comment:
this is the WordPress @since `meta_form()` in `wp-admin/includes/template.php` not sure we should use it?